### PR TITLE
[minor_change] Add support for required attributes introduced in ndo4…

### LINKED ISSUE #377: Binding type is required in ndo 4.2.1 but not option are available in ansible module
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Binding type is required in ndo 4.2.1 but not option are available in ansible module

### Affected Module Name(s):

cisco.mso.mso_schema_site_anp_epg_domain

### MSO version and MSO Platform

V 4.2.1

### APIC version and APIC Platform for Site Level Resources

V 5.2.7

### Collection versions

cisco.mso 2.2.1

### Output/ Error message

MSO Error 400: site-local EPG XX in site XX in template XX: BindingType is required for VMware VMM Domain

### Expected Behavior

BindingType should be put to default because it's required in ndo now.

### Actual Behavior

BindingType is not available and no default value is put so we get an error message.

### Playbook tasks to Reproduce

Add VMM domain binding to EPG

### References

* #0000

### PR DESCRIPTION
**Pull request by akinross**
….2 for mso_schema_site_anp_epg_domain

fixes #377

### CODE REVIEW

**Review by samiib — DISMISSED** [review]
LGTM

**Review by shrsr — APPROVED** [review]
LGTM

**Review by gmicol — DISMISSED** [review]
LGTM

File: plugins/modules/mso_schema_site_anp_epg_domain.py [hunk 1]
```diff
@@ -109,6 +113,55 @@
     description:
     - Distinguished name of EPG lagpolicy. This attribute can only be used with vmmDomain domain association.
     type: str
+  delimiter:
+    description:
+    - Which delimiter to use with this domain association. This attribute can only be used with vmmDomain domain association.
```
Comment (lhercot) [inline plugins/modules/mso_schema_site_anp_epg_domain.py:118]:
```suggestion
    - Which delimiter to use with this domain association. This attribute can only be used with VMM domain associations.
```

File: plugins/modules/mso_schema_site_anp_epg_domain.py [hunk 2]
```diff
@@ -109,6 +113,55 @@
...
+    type: str
+    choices: [ '+', '|', '~', '!', '@', '^', '=' ]
+  binding_type:
+    description:
+    - Which binding_type to use with this domain association. This attribute can only be used with vmmDomain domain association.
```
Comment (lhercot) [inline plugins/modules/mso_schema_site_anp_epg_domain.py:123]:
```suggestion
    - Which binding_type to use with this domain association. This attribute can only be used with VMM domain associations.
```

Comment (lhercot) [reply plugins/modules/mso_schema_site_anp_epg_domain.py:123]: Same for all others

File: plugins/modules/mso_schema_site_anp_epg_domain.py [hunk 3]
```diff
@@ -109,6 +113,55 @@
...
+    type: int
+  port_allocation:
+    description:
+    - Port allocation for the binding type. This attribute can only be used with vmmDomain domain association and binding type in static.
+    - Required when O(binding_type=static).
+    type: str
+    choices: [ elastic, fixed ]
+  netflow_pref:
+    description:
+    - Which netflow_pref to use with this domain association. This attribute can only be used with vmmDomain domain association.
```
Comment (lhercot) [inline plugins/modules/mso_schema_site_anp_epg_domain.py:139]: Which? It is only enable/disabled so it is not really which?

**Review by lhercot — CHANGES_REQUESTED** [review]

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by anvitha-jain — APPROVED** [review]

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**